lucene/core/src/java/org/apache/lucene/util/automaton/SortedIntSet.java

```diff
@@ -151,126 +141,23 @@ public void computeHash() {
     }
   }
 
+  /**
+   * Create a FrozenIntSet from the current values in this IntSet.
+   *
+   * Note: Must call computeHash() before calling this method
+   *
+   * @param state the state to save
+   * @return a FrozenIntSet that has the same values and hashCode as this set
+   */
   public FrozenIntSet freeze(int state) {
     final int[] c = new int[upto];
     System.arraycopy(values, 0, c, 0, upto);
```

Comment:
isn't there a futurearrays method that does this in a single step (without intermediate zeroed int[])?

Reply:
Arrays.copyOf does it in two steps, same as here. FutureArrays has compare, equals, and mismatch that I see?

Reply:
You look at Java code but the difference is that Array.copyOf is (I believe) an jvm intrinsic so it should be replaced with a more optimized code. The pair of allocation+arraycopy should as well be optimized for that matter but it requires execution graph analysis and copyOf is straightforward.

Reply:
Interestingly enough, forbidden APIs doesn't like Arrays.copyOf... I think we're fine here though.